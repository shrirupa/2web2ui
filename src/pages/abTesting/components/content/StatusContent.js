import React from 'react';
import { UnstyledLink } from '@sparkpost/matchbox';

const StatusContent = ({ test, rescheduling }) => (
  <div>
    {
      (test.status === 'draft' || rescheduling) && <div><p>You can update your Test Name and select a start time and end time to schedule your A/B Test.</p>
      </div>
    }
    {
      test.status === 'scheduled' && <div>
        <p>This test has been scheduled. You can update your Test Name, start time, or end time.</p>
      </div>

    }
    {
      test.status === 'running' && <div>
        <p>This test is currently running. It will conclude when either:</p>
        <ul>
          <li>the end date + engagement timeout period is reached</li>
          {test.test_mode === 'bayesian' && <li>the desired confidence level has been reached</li>}
          <li>the desired sample size limit is reached + engagement timeout period is reached</li>
        </ul>
      </div>

    }
    {
      test.status === 'cancelled' && !rescheduling && <div>
        <p>This test run was cancelled. To re-run this test, click "Edit and Reschedule Test".</p>
      </div>

    }
    {
      test.status === 'completed' && !rescheduling && test.test_mode === 'learning' && <p>This test has concluded and it now sending the default template. You may override the default with a new one without starting a new test. This test was started and completed at the times shown.</p>
    }
    {
      test.status === 'completed' && !rescheduling && test.test_mode === 'bayesian' && test.winning_template_id !== test.default_template.template_id && <p>This test has concluded and it now sending the winning template. You may override it with a new one without starting a new test. This test was started and completed at the times shown.</p>
    }
    {
      test.status === 'completed' && !rescheduling && test.test_mode === 'bayesian' && test.winning_template_id === test.default_template.template_id && <p>This test has concluded and it now sending the default template. You may override it with a new one without starting a new test. This test was started and completed at the times shown.</p>
    }
    {
      (test.status !== 'completed' || rescheduling) && <p><span>You can read about the different A/B Test states </span><UnstyledLink external to='https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states'>here</UnstyledLink>.</p>
    }
  </div>
);

export default StatusContent;
