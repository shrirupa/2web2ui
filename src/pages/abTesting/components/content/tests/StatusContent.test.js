import { shallow } from 'enzyme';
import React from 'react';
import StatusContent from '../StatusContent';

describe('Status Content Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      test: {}
    };
    wrapper = shallow(<StatusContent {...props} />);
  });

  it('should render nothing when a test is malformed', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render draft-specific help content when test is in draft mode', () => {
    wrapper.setProps({ test: { status: 'draft' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render scheduled-specific help content when test is in scheduled mode', () => {
    wrapper.setProps({ test: { status: 'scheduled' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render running-specific help content when test is in running mode', () => {
    wrapper.setProps({ test: { status: 'running' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render cancelled-specific help content when test is in cancelled mode', () => {
    wrapper.setProps({ test: { status: 'cancelled' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render distinct help content when test is completed for learning mode tests', () => {
    wrapper.setProps({ test: { status: 'completed', test_mode: 'learning' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render distinct help content when test is completed for bayesian mode tests and the winner was found', () => {
    wrapper.setProps({ test: { status: 'completed', test_mode: 'bayesian', winning_template_id: 'one', default_template: { template_id: 'two' }}});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render distinct help content when test is completed for bayesian mode tests and the winner was NOT found', () => {
    wrapper.setProps({ test: { status: 'completed', test_mode: 'bayesian', winning_template_id: 'one', default_template: { template_id: 'one' }}});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render rescheduling-specific help content when test is in completed mode and rescheduling the test', () => {
    wrapper.setProps({ test: { status: 'completed' }, rescheduling: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render rescheduling-specific help content when test is in cancelled mode and rescheduling the test', () => {
    wrapper.setProps({ test: { status: 'cancelled' }, rescheduling: true });
    expect(wrapper).toMatchSnapshot();
  });
});
