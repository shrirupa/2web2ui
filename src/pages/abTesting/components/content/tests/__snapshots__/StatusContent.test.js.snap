// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Content Component should render cancelled-specific help content when test is in cancelled mode 1`] = `
<div>
  <div>
    <p>
      This test run was cancelled. To re-run this test, click "Edit and Reschedule Test".
    </p>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render distinct help content when test is completed for bayesian mode tests and the winner was NOT found 1`] = `
<div>
  <p>
    This test has concluded and it now sending the default template. You may override it with a new one without starting a new test. This test was started and completed at the times shown.
  </p>
</div>
`;

exports[`Status Content Component should render distinct help content when test is completed for bayesian mode tests and the winner was found 1`] = `
<div>
  <p>
    This test has concluded and it now sending the winning template. You may override it with a new one without starting a new test. This test was started and completed at the times shown.
  </p>
</div>
`;

exports[`Status Content Component should render distinct help content when test is completed for learning mode tests 1`] = `
<div>
  <p>
    This test has concluded and it now sending the default template. You may override the default with a new one without starting a new test. This test was started and completed at the times shown.
  </p>
</div>
`;

exports[`Status Content Component should render draft-specific help content when test is in draft mode 1`] = `
<div>
  <div>
    <p>
      You can update your Test Name and select a start time and end time to schedule your A/B Test.
    </p>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render nothing when a test is malformed 1`] = `
<div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render rescheduling-specific help content when test is in cancelled mode and rescheduling the test 1`] = `
<div>
  <div>
    <p>
      You can update your Test Name and select a start time and end time to schedule your A/B Test.
    </p>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render rescheduling-specific help content when test is in completed mode and rescheduling the test 1`] = `
<div>
  <div>
    <p>
      You can update your Test Name and select a start time and end time to schedule your A/B Test.
    </p>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render running-specific help content when test is in running mode 1`] = `
<div>
  <div>
    <p>
      This test is currently running. It will conclude when either:
    </p>
    <ul>
      <li>
        the end date + engagement timeout period is reached
      </li>
      <li>
        the desired sample size limit is reached + engagement timeout period is reached
      </li>
    </ul>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;

exports[`Status Content Component should render scheduled-specific help content when test is in scheduled mode 1`] = `
<div>
  <div>
    <p>
      This test has been scheduled. You can update your Test Name, start time, or end time.
    </p>
  </div>
  <p>
    <span>
      You can read about the different A/B Test states 
    </span>
    <UnstyledLink
      external={true}
      to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/#ab-testing-states"
    >
      here
    </UnstyledLink>
    .
  </p>
</div>
`;
